src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -1,95 +1,92 @@
+        if (listResourcesInGroupResponse.response.body.contains)
+          this.setStateFromServerResponse(groupName, listResourcesInGroupResponse.response.body.contains)
+      })
+    } catch(error) {
+      const errors = [...this.state.errors, error]
+      this.setState({ errors: errors })
+    }
   }
 
-  fulfillGroupData = (data) => {
-    const groupName = resourceToName(data.response.body['@id'])
+  setStateFromServerResponse = (groupName, groupContains) => {
+    // Array-ify the value of `groupContains` as it can be either a string or an array
+    const templateIds = [].concat(groupContains)
 
-    if (data.response.body.contains !== undefined) {
-      const contains = [].concat(data.response.body.contains)
+    templateIds.forEach(async templateId => {
+      const templateName = resourceToName(templateId)
+      const templateResponse = await getResourceTemplate(templateName, groupName)
+      const resourceTemplateBody = templateResponse.response.body
 
-      contains.map(c => {
-        const name = resourceToName(c)
+      const retrievedTemplateObject = {
+        key: resourceTemplateBody.id,
+        name: resourceTemplateBody.resourceLabel,
+        uri: templateId,
+        id: resourceTemplateBody.id,
+        group: groupName
+      }
 
-        this.groupDataPromise(groupName, name).then((data) => {
-          const rt = data.response.body
-          this.setState({tempState: {key: rt.id, name: rt.resourceLabel, uri: c, id: rt.id, group: groupName}})
-          const joined = this.state.templatesForGroup.slice(0)
-          if (!_.find(joined, ['key', this.state.tempState.key])) {
-            joined.push(this.state.tempState)
-          }
-          this.setState({templatesForGroup: joined})
-        }).catch(() => {})
-      })
-    }
-
-  }
+      const templates = [...this.state.resourceTemplates]
 
-  groupDataPromise = (groupName, name) => new Promise(async (resolve) => {
-    await resolve(getResourceTemplate(name, groupName))
-  })
+      // Check if newly retrieved template is already stored in component state. If not, add it.
+      if (!templates.some(template => template.key === retrievedTemplateObject.key))
+        templates.push(retrievedTemplateObject)
 
+      this.setState({
+        resourceTemplates: templates
+      })
+    })
+  }
 
   linkFormatter = (cell, row) => {
     return(
       <Link to={{pathname: '/editor', state: { resourceTemplateId: row.id }}}>{cell}</Link>
     )
   }
 
-  render(){
-    if (this.state.message) {
+  render() {
+    if (this.state.errors.length > 0) {
       return (
         <div className="alert alert-warning alert-dismissible">
           <button className="close" data-dismiss="alert" aria-label="close">&times;</button>
-          No connection to the Sinopia Server is available, or there are no resources for any group
+          No connection to the Sinopia Server is available, or there are no resources for any group.
```

Comment:
Not a blocking comment but it seems this should be two separate error messages; 
1.  one for when there is no connection to the Sinopia Server
1.  another message if there are no resources for any group.

Reply:
I totally agree! would love to see this ticketed and taken on. (for now, inclined not to address this in this already sizable branch since it's not connected to the issues I'm working on.)

Reply:
Done, now being tracked in issue [352](https://github.com/LD4P/sinopia_editor/issues/532)